server/protocols/protocols-imap4/src/main/java/org/apache/james/imapserver/netty/IMAPServerFactory.java

```diff
@@ -76,16 +78,19 @@ protected IMAPServer createServer(HierarchicalConfiguration<ImmutableNode> confi
     protected List<AbstractConfigurableAsyncServer> createServers(HierarchicalConfiguration<ImmutableNode> config) throws Exception {
         List<AbstractConfigurableAsyncServer> servers = new ArrayList<>();
         List<HierarchicalConfiguration<ImmutableNode>> configs = config.configurationsAt("imapserver");
-        
+
         for (HierarchicalConfiguration<ImmutableNode> serverConfig: configs) {
             IMAPServer server = createServer(serverConfig);
             server.setFileSystem(fileSystem);
             server.configure(serverConfig);
             servers.add(server);
         }
+        imapServers = servers.stream().map(IMAPServer.class::cast).toList();
 
         return servers;
-
     }
 
+    public List<IMAPServer> getImapServers() {

```

Comment:
We can use https://github.com/apache/james-project/blob/master/server/protocols/protocols-library/src/main/java/org/apache/james/protocols/lib/netty/AbstractServerFactory.java#L66 and filter by IMAP server type.

![image](https://github.com/user-attachments/assets/7489396c-25b4-49b6-8f1b-a13bb453b458)

Reply:
Yes it would prevent the uggly field allocation, +1 quan

Reply:
Fixed. Sr I did not know that the attribute servers has already existed.

Reply:
BTW what prevent us from doing the cast within IMAPHealthCheck ?

This would prevent a change here...

Reply:
Yes we can. However, I don't see any problems with "a change" in IMAPServerFactory.